server/game/cards/01_SOR/units/BobaFettDisintegrator.ts

```diff
@@ -0,0 +1,39 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { StateWatcherRegistrar } from '../../../core/stateWatcher/StateWatcherRegistrar';
+import { CardsEnteredPlayThisPhaseWatcher } from '../../../stateWatchers/CardsEnteredPlayThisPhaseWatcher';
+
+export default class BobaFettDisintegrator extends NonLeaderUnitCard {
+    // initiate watcher record
+    private cardsEnteredPlayThisPhaseWatcher: CardsEnteredPlayThisPhaseWatcher;
+
+    protected override getImplementationId() {
+        return {
+            id: '4156799805',
+            internalName: 'boba-fett#disintegrator'
+        };
+    }
+
+    // setup watcher
+    protected override setupStateWatchers(registrar: StateWatcherRegistrar) {
+        this.cardsEnteredPlayThisPhaseWatcher = AbilityHelper.stateWatchers.cardsEnteredPlayThisPhase(registrar, this);
+    }
+
+    public override setupCardAbilities() {
+        this.addOnAttackAbility({
+            title: 'If this unit is attacking an exhausted unit that didn\'t enter play this round, deal 3 damage to the defender.',
+            immediateEffect: AbilityHelper.immediateEffects.conditional((attackContext) => ({
+                condition: () => {
+                    // we find the cards that the opponent played this phase and check if it's the card targeted
+                    return this.cardsEnteredPlayThisPhaseWatcher.getCardsPlayed((playedCardEntry) =>
+                        playedCardEntry.playedBy === attackContext.source.activeAttack.target.owner && attackContext.source.activeAttack.target === playedCardEntry.card).length === 0 &&
```

Comment:
Missed this before - since `activeAttack` is nullable we should probably do `attackContext.source.activeAttack?.target.owner` with the `?` here (same for other uses of the property in this statement).

It would conceivably get set to null if the attacker were defeated before this ability triggered, for example, b/c it gets forcibly set to null if the card is moved out of the arena.

Reply:
added optional ? to activeAttack since it can be nullable in BobaFettDisintegrator.ts.